rts/lra/coordinator/src/main/java/io/narayana/lra/coordinator/api/Coordinator.java

```diff
@@ -113,107 +116,131 @@ public Coordinator() {
             @Parameter(name = STATUS_PARAM_NAME, description = "Filter the returned LRAs to only those in the give state (see CompensatorStatus)")
-            @QueryParam(STATUS_PARAM_NAME) @DefaultValue("") String state) {
+            @QueryParam(STATUS_PARAM_NAME) @DefaultValue("") String state,
+            @Parameter(ref = LRAConstants.LRA_API_VERSION_HEADER_NAME)
+            @HeaderParam(LRAConstants.LRA_API_VERSION_HEADER_NAME) @DefaultValue(NARAYANA_LRA_API_VERSION_STRING) String version) {
         LRAStatus requestedLRAStatus = null;
-        if(!state.isEmpty()) {
-            requestedLRAStatus = LRAStatus.valueOf(state);
+        if (!state.isEmpty()) {
+            try {
+                requestedLRAStatus = LRAStatus.valueOf(state);
+            } catch (IllegalArgumentException e) {
+                String errorMsg = "Status " + state + " is not a valid LRAStatus value";
+                throw new WebApplicationException(errorMsg, e,
+                        Response.status(BAD_REQUEST).header(LRA_API_VERSION_HEADER_NAME, version).entity(errorMsg).build());
+            }
         }
 
-        Collection<LRAData> lras = lraService.getAll(requestedLRAStatus);
-
-        if (lras == null) {
-            LRALogger.i18NLogger.error_invalidQueryForGettingLraStatuses(state);
-            String errMsg = String.format("Invalid query '%s' to get LRAs", state);
-            throw new WebApplicationException(errMsg,
-                    Response.status(BAD_REQUEST).entity(errMsg).build());
-        }
+        List<LRAData> lras = lraService.getAll(requestedLRAStatus);
 
-        return lras;
+        return Response.ok()
+            .entity(lras)
+            .header(LRA_API_VERSION_HEADER_NAME, version).build();
     }
 
     @GET
     @Path("{LraId}/status")
     @Produces(MediaType.TEXT_PLAIN)
     @Operation(summary = "Obtain the status of an LRA as a string")
-    @Schema(implementation = String.class)
     @APIResponses({
-        @APIResponse(responseCode = "404",
-            description = "The coordinator has no knowledge of this LRA"),
-        @APIResponse(responseCode = "204",
-            description = "The LRA exists and has not yet been asked to close or cancel"),
-        @APIResponse(responseCode = "200",
-            description = "The LRA exists. The status is reported in the content body.")
+        @APIResponse(responseCode = "200", description = "The LRA exists. The status is reported in the content body.",
+            content = @Content(schema = @Schema(implementation = String.class)),
+            headers = { @Header(ref = LRAConstants.LRA_API_VERSION_HEADER_NAME) }),
+        @APIResponse(responseCode = "404", description = "The coordinator has no knowledge of this LRA",
+            content = @Content(schema = @Schema(implementation = String.class))),
+        @APIResponse(responseCode = "417", description = "The requested version provided in HTTP Header is not supported by this end point",
```

Comment:
Adding an extra level of indirection just makes it that much harder for the reader, following the link to get the value and then returning causes the reader to loose context.

There are search and replace tools for changing it.

And we wouldn't redefine JAX-RS constants with our own, we would use Response.Status.EXPECTATION_FAILED so it would not help.

Reply:
For now I'm leaving it as it is.